pyro/nn/auto_reg_nn.py

```diff
@@ -2,11 +2,75 @@
 
 import warnings
 
+import numpy as np
```

Comment:
as mentioned elsewhere, please remove all numpy usage

Reply:
Okay, I've removed all NumPy usage and made the requested changes